### DISCUSSION TITLE
connecting to the hosted foxglove via API

### DISCUSSION THREAD

**Discussion by puzarsofts** [foxglove/community#1023] [Q&A]
Hello

I am trying to connect to foxglove hosted on the kubernetes via API, I have created token but .... I cannot connect to the instance, is this ok that I am only providing api as parameter? Shouldn't I anyhow point the instance ?

**Comment by defunctzombie** [comment]
Are you providing your API secret as a bearer token as described here: https://docs.foxglove.dev/api#section/Authentication ?

**Comment by puzarsofts** [comment]
I am using python library, where I do not see any place to add it

**Comment by defunctzombie** [comment]
Have a look at the examples here: https://github.com/foxglove/foxglove-python/tree/main/examples

**Comment by puzver1** [comment]
That doesn't help at all, I can type token, and that's it, and if I do that - I do receive

`Traceback (most recent call last):
  File "/Users/lkmn/code/abc/foxgloveapi/foxgloveapi.py", line 7, in <module>
    client.get_devices()
  File "/Users/lkmn/code/abc/foxgloveapi/.venv/lib/python3.9/site-packages/foxglove/client.py", line 539, in get_devices
    json = json_or_raise(response)
  File "/Users/lkmn/code/abc/foxgloveapi/.venv/lib/python3.9/site-packages/foxglove/client.py", line 147, in json_or_raise
    response.raise_for_status()
  File "/Users/lkmn/code/abc/foxgloveapi/.venv/lib/python3.9/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.foxglove.dev/v1/devices`

**Comment by puzver1** [comment]
ok, please ignore my previous queries - while working directly with API everything works fine, issue is on foxglove-client side - we can close topic
